dlt-connector/src/controller/Community.ts

```diff
@@ -0,0 +1,36 @@
+import { CommunityDraft } from '@/graphql/input/CommunityDraft'
+import { uuid4ToBuffer } from '@/utils/typeConverter'
+import { Community } from '@entity/Community'
+import { getDataSource } from '@typeorm/DataSource'
+import { crypto_generichash as cryptoHash } from 'sodium-native'
+
+export const iotaTopicFromCommunityUUID = (communityUUID: string): string => {
+  const hash = Buffer.alloc(32)
+  cryptoHash(hash, uuid4ToBuffer(communityUUID))
+  return hash.toString('hex')
+}
+
+export const isExist = async (community: CommunityDraft | string): Promise<boolean> => {
+  const iotaTopic =
+    community instanceof CommunityDraft ? iotaTopicFromCommunityUUID(community.uuid) : community
+  const result = await getDataSource().manager.find(Community, {
+    where: { iotaTopic },
+  })
+  return result.length > 0
+}
+
+export const create = (community: CommunityDraft, topic?: string): Community => {
+  const communityEntity = new Community()
```

Comment:
Isn't it necessary to use the internal EntityRepository with internal TX-Support to instanciate an entity by calling here Community.create() than the new-Constructor?
Then you can use for writing later the expression Community.save(comEntity), which handles the transaction and differ between insert and update

Reply:
If you want to use the Internal EntityRepository yes, also the entity classes must then be derived from BaseEntity. I have changed this now, originally I saved this optimization to be a bit faster.